documentation/query-api.md

```diff
@@ -0,0 +1,213 @@
+returns status info for a single service), rather than the `/services` endpoint
+(which aggregates status for all registered services).
+
+### Examples
+
+Using `curl` from localhost:
+
+Get the service status of all registered services in an application:
+
+    curl -k https://localhost:8000/status/v1/services
+
+    {
+        "puppet-server": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 1,
+            "service_version": "1.0.9-SNAPSHOT",
+            "status": {
+                "bar": "bar",
+                "foo": "foo"
+            }
+        },
+
+        "other-service": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 2,
+            "service_version": "0.0.1-SNAPSHOT",
+            "status": {
+                "baz": [1, 2, 3],
+                "bang": {"key": "value"}
+            }
+        }
+    }
+
+Get the service status of all registered services in an application, at a
+ specified level of detail:
+
+    curl -k "https://localhost:8140/status/v1/services?level=critical"
+
+    {
+        "puppet-server": {
+            "detail_level": "critical",
+            "is_running": "true",
+            "service_status_version": 1,
+            "service_version": "1.0.9-SNAPSHOT",
+            "status": null
+        },
+
+        "other-service": {
+            "detail_level": "info",
+            "is_running": "true",
+            "service_status_version": 2,
+            "service_version": "0.0.1-SNAPSHOT",
+            "status": null
+        }
+    }
+
+## `GET /status/v1/services/<service-name>`
+
+This will return status information for a single, specified service from the running
+application.
+
+### URL Parameters
+
+* `level`: Optional.  A JSON String from among the legal
+[Status Detail Levels](#StatusDetailLevel) listed above.  Status information for
+the requested service will be provided at the requested level of detail.  If
+not provided, the default level is `"info"`.
+
+* `service-status-version`: Optional.  A JSON integer specifying the desired status
+format version for the requested service.  If not provide, defaults to the latest
+available status format version for the service.
+
+### Response Format
+
+The response format will be a JSON _Object_ with a single entry,
+which will look something like this:
+
+    {<service-name>: {
+        "service_version": <service-version>,
+        "service_status_version": <service-status-version>,
+        "detail_level": <detail-level>,
+        "is_running": <service-state>,
+        "status": <any>
+        }
+    }
+
+For detailed information, please see the [Wire Format Specification](./wire-formats.md).
+
+### Examples
+
+Using `curl` from localhost:
+
+Get the service status for a specified service in the application:
+
+    curl -k https://localhost:8000/status/v1/services/other-service
+
+    {
+        "other-service": {
```

Comment:
Yeah, I think all of the keys should use underscores (instead of hyphens) to conform to our style guide.